naval_serial_bridge:
  ros__parameters:
    serial_port: "auto"
    baud_rate: 115200
    reconnect_interval: 2.0

naval_web_server:
  ros__parameters:
    port: 5000
    host: "0.0.0.0"
    rosbridge_port: 9090
    cam_server_port: 8090
    cmd_vel_topic: "/cmd_vel_raw"

naval_obstacle_avoidance:
  ros__parameters:
    obstacle_threshold: 0.20
    clear_threshold: 0.30
    front_angle: 30.0
    reverse_speed: -0.10
    reverse_distance: 0.12
    turn_speed_max: 0.4
    turn_speed_min: 0.15
    turn_proportional_gain: 0.5
    stop_duration: 0.1
    pause_after_reverse: 0.2
    scan_duration: 0.3
    turn_min_duration: 0.3
    turn_max_duration: 5.0
    max_avoidance_attempts: 5
    reverse_timeout: 2.0
    lidar_timeout: 0.5
    lidar_angle_offset: 165.0
    min_valid_range: 0.18
